op-e2e/erigon.go

```diff
@@ -0,0 +1,157 @@
+package op_e2e
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/stretchr/testify/require"
+)
+
+func BuildErigon(buildPath string) error {
+	cmd := exec.Command("go", "build", "-o", buildPath, "github.com/ledgerwatch/erigon/cmd/erigon")
+	cmd.Dir = filepath.Join("..", "op-erigon")
+	sess, err := gexec.Start(cmd, os.Stdout, os.Stderr)
+	if err != nil {
+		return err
+	}
+	gm := gomega.NewGomega(func(msg string, _ ...int) {
+		err = errors.New(msg)
+	})
+	gm.Eventually(sess, time.Minute).Should(gexec.Exit(0))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type ErigonRunner struct {
+	Name     string
+	BinPath  string
+	DataDir  string
+	JWTPath  string
+	ChainID  uint64
+	GasCeil  uint64
+	GasPrice uint64
+	Genesis  *core.Genesis
+}
+
+func (er *ErigonRunner) Run(t *testing.T) ErigonInstance {
+	if er.BinPath == "" {
+		t.Error("no erigon bin path set")
+	}
+
+	if er.DataDir == "" {
+		er.DataDir = t.TempDir()
+	}
+
+	if er.JWTPath == "" {
+		er.JWTPath = writeDefaultJWT(t)
+	}
+
+	if er.ChainID == 0 {
+		er.ChainID = 901
+	}
+
+	if er.Genesis == nil {
+		er.Genesis = &core.Genesis{
+			Alloc: core.GenesisAlloc{
+				common.Address{1}: core.GenesisAccount{Balance: big.NewInt(1)},
+			},
+			Config:     &params.ChainConfig{ChainID: big.NewInt(901)},
+			Difficulty: big.NewInt(0),
+		}
+	}
+
+	genesisPath := filepath.Join(er.DataDir, "genesis.json")
+	o, err := os.Create(genesisPath)
+	require.NoError(t, err)
+	err = json.NewEncoder(o).Encode(er.Genesis)
+	require.NoError(t, err)
+
+	gt := gomega.NewWithT(t)
+	cmd := exec.Command(
+		er.BinPath,
+		"--datadir", er.DataDir,
+		"init", genesisPath,
+	)
+	sess, err := gexec.Start(
+		cmd,
+		gexec.NewPrefixedWriter(er.Name, os.Stdout),
+		gexec.NewPrefixedWriter(er.Name, os.Stderr),
+	)
+	gt.Expect(err).NotTo(gomega.HaveOccurred())
+	gt.Eventually(sess.Err, time.Minute).Should(gbytes.Say("Successfully wrote genesis state"))
```

Comment:
This sort of construct is why gomega is being pulled in, as this one line will block execution, polling every 10ms, waiting for a particular substring from the process output.  It can obviously be done by spinning up another go routine, creating channels, etc. but, the simplicity and readability make it worth the import for me.